### PR TITLE
Revert rest change to ChateauRequest.

### PR DESCRIPTION
**Pull request by phulin**
A change in #1137 ([here](https://github.com/kolmafia/kolmafia/commit/db6f43f0562286bc9eb2013c509663fc1052221b#diff-4e27d487224bb81a99bdc43788d64468183b9e44d789954232042a5c3062c9b3L116-R115)) caused Chateau rests to register incorrectly if users click on some parts of the bed image in the Chateau.

In particular, clicking on the image of the bed triggers a request with `action=chateau_restbox` no matter whether the rest is free or not, so clicking on the bed will incorrectly set free rests to the maximum. Clicking on the "Rest in your Bed" text uses `action=chateau_restlabelfree` if the rest is free.

@Veracity0 Flagging that this reverts a change from a few days ago - not sure why that change was made originally.

### CODE REVIEW

**Comment by phulin** [comment]
Will write a test for this... stay tuned.

File: src/net/sourceforge/kolmafia/request/GenericRequest.java [hunk 1]
```diff
@@ -1394,11 +1394,11 @@ public void externalExecute() {
   public static final boolean shouldIgnore(final GenericRequest request) {
     String requestURL = GenericRequest.decodeField(request.formURLString);
     return requestURL == null
-        ||
+        //        ||
```
Comment (Veracity0) [inline src/net/sourceforge/kolmafia/request/GenericRequest.java:1397]: Absolutely not. Remove your private change. :)

File: test/net/sourceforge/kolmafia/request/ChateauRequestTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+public class ChateauRequestTest {
+  @BeforeAll
+  public static void beforeAll() {
+    KoLCharacter.reset(true);
+    KoLCharacter.reset("ChateauRequestTest");
+    Preferences.reset("ChateauRequestTest");
+    Preferences.setBoolean("chateauAvailable", true);
```
Comment (Veracity0) [inline test/net/sourceforge/kolmafia/request/ChateauRequestTest.java:22]: Do these using withProperty() Cleanups

**Review by Veracity0 — CHANGES_REQUESTED** [review]

File: src/net/sourceforge/kolmafia/request/GenericRequest.java (hunk 1 above)
Comment (gausie) [reply src/net/sourceforge/kolmafia/request/GenericRequest.java:1397]: Luckily this will cause a test I added to fail

Comment (Veracity0) [inline src/net/sourceforge/kolmafia/request/GenericRequest.java]: Absolutely not. You may not submit these private changes.

File: test/net/sourceforge/kolmafia/request/ChateauRequestTest.java (hunk 2 above)
Comment (Veracity0) [inline test/net/sourceforge/kolmafia/request/ChateauRequestTest.java:22]: Do these using withProperty() Cleanups.

**Review by Veracity0 — CHANGES_REQUESTED** [review]
The test failures were looking right at your private changes. ;)

File: src/net/sourceforge/kolmafia/request/GenericRequest.java (hunk 1 above)
Comment (phulin) [reply src/net/sourceforge/kolmafia/request/GenericRequest.java]: usually I don't make this mistake... thank you for adding that test, Sam :)

Comment (phulin) [reply src/net/sourceforge/kolmafia/request/GenericRequest.java]: Fixed.

File: test/net/sourceforge/kolmafia/request/ChateauRequestTest.java (hunk 2 above)
Comment (phulin) [reply test/net/sourceforge/kolmafia/request/ChateauRequestTest.java]: Made an attempt, let me know if this is the pattern you were thinking of.

Comment (Veracity0) [reply test/net/sourceforge/kolmafia/request/ChateauRequestTest.java]: Here's what I would do.

ChateauRequestTest is for testing everything about the Chateau.
You are testing free rests in particular.

The pattern I would like is this:

```
public class ChateauRequestTest {

  @BeforeAll
  public static void beforeAll() {
    KoLCharacter.reset("ChateuRequestTest");
    Preferencesr.reset("ChateuRequestTest");
  }

  @BeforeEach
  public void beforeEach() {
    // stuff
  }

  @Nested
  class FreeRest {

    public Cleanups propertyCleanups() {
      // ... stuff that applies to free rests
      return cleanups;
    }

    @Test
    public void test1() {
      var cleanups = new Cleanups(propertyCleanups());
      try (cleanups) {
        // test code
      }
    }
  }
}
```

**Review by gausie — DISMISSED** [review]

File: test/net/sourceforge/kolmafia/request/ChateauRequestTest.java (hunk 2 above)
Comment (phulin) [reply test/net/sourceforge/kolmafia/request/ChateauRequestTest.java]: Restructured the test - I think this is the right way to do it. Let me know if you disagree.

Comment (phulin) [reply test/net/sourceforge/kolmafia/request/ChateauRequestTest.java]: (And thanks for the help - it's been at least a decade since I used JUnit...)

Comment (Veracity0) [reply test/net/sourceforge/kolmafia/request/ChateauRequestTest.java]: Very nice!

**Review by Veracity0 — APPROVED** [review]

**Pull request merged**